src/swappers/TradeFactorySwapper.sol

```diff
@@ -6,101 +6,154 @@ import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol
+     * @notice Get the current Trade Factory.
+     * @dev We use a getter so trade factory can only be set through the
+     *   proper functions to avoid issues.
+     * @return The current trade factory in use if any.
+     */
     function tradeFactory() public view returns (address) {
         return _tradeFactory;
     }
 
+    /**
+     * @notice Get the current tokens being sold through the Trade Factory.
+     * @dev We use a getter so the array can only be set through the
+     *   proper functions to avoid issues.
+     * @return The current array of tokens being sold if any.
+     */
     function rewardTokens() public view returns (address[] memory) {
         return _rewardTokens;
     }
 
+    /**
+     * @dev Add an array of tokens to sell to its corresponding `_to_.
+     */
     function _addTokens(address[] memory _from, address[] memory _to) internal {
         for (uint256 i; i < _from.length; ++i) {
             _addToken(_from[i], _to[i]);
         }
     }
 
+    /**
+     * @dev Add the `_tokenFrom` to be sold to `_tokenTo` through the Trade Factory
+     */
     function _addToken(address _tokenFrom, address _tokenTo) internal {
-        address tradeFactory_ = _tradeFactory;
-        if (tradeFactory_ != address(0)) {
-            ERC20(_tokenFrom).safeApprove(tradeFactory_, type(uint256).max);
+        address _tf = tradeFactory();
+        if (_tf != address(0)) {
+            ERC20(_tokenFrom).safeApprove(_tf, type(uint256).max);
 
-            ITradeFactory(tradeFactory_).enable(_tokenFrom, _tokenTo);
+            ITradeFactory(_tf).enable(_tokenFrom, _tokenTo);
         }
 
         _rewardTokens.push(_tokenFrom);
     }
 
+    /**
+     * @dev Remove a specific `_tokenFrom` that was previously added to not be
+     * sold through the Trade Factory any more.
+     */
     function _removeToken(address _tokenFrom, address _tokenTo) internal {
-        address[] memory rewardTokens_ = _rewardTokens;
-        for (uint256 i; i < rewardTokens_.length; ++i) {
-            if (rewardTokens_[i] == _tokenFrom) {
-                if (i != rewardTokens_.length - 1) {
+        address _tf = tradeFactory();
+        address[] memory _rewardTokensLocal = rewardTokens();
+        for (uint256 i; i < _rewardTokensLocal.length; ++i) {
+            if (_rewardTokensLocal[i] == _tokenFrom) {
+                if (i != _rewardTokensLocal.length - 1) {
                     // if it isn't the last token, swap with the last one/
-                    rewardTokens_[i] = rewardTokens_[rewardTokens_.length - 1];
+                    _rewardTokensLocal[i] = _rewardTokensLocal[
+                        _rewardTokensLocal.length - 1
+                    ];
                 }
-                ERC20(_tokenFrom).safeApprove(_tradeFactory, 0);
-                ITradeFactory(_tradeFactory).disable(_tokenFrom, _tokenTo);
+                ERC20(_tokenFrom).safeApprove(_tf, 0);
+                ITradeFactory(_tf).disable(_tokenFrom, _tokenTo);
```

Comment:
To fix [this](https://github.com/yearn/yearn-strategies/issues/575#issuecomment-2024941589) issue we need to add:
```solidity
if (_tf != address(0)) ERC20(_tokenFrom).safeApprove(_tf, 0);
```

Would that be a problem for the other actions that the trade factory will be used for?

Reply:
added